src/app/layouts/ModalWindow.js

```diff
@@ -0,0 +1,34 @@
+export class ModalWindow {
+  constructor() {
+    const modalWindow = document.createElement('div');
+    modalWindow.classList.add('modalWindow');
+    modalWindow.id = 'modalWindow';
+
+    const parent = document.getElementById('swquiz-app');
+    parent.appendChild(modalWindow);
+  }
+
+  show(content) {
+    // ""Komponent umożliwia wyświetlenie okna modalnego z zawartością przekazaną jako inny komponent na wejściu do funkcji.""
+    // wiadomo jaki będzie typ komponentu jaki przyjdzie? Bo if (content) to chyba średni warunek...
+    if (content) {
+      content.classList.add('modalWindow__content');
+
+      // create background for content
+      const contentBackground = document.createElement('div');
+      contentBackground.classList.add('modalWindow__contentBackground');
+      contentBackground.id = 'modalWindow-bg';
+      modalWindow.appendChild(contentBackground);
+
+      contentBackground.appendChild(content);
+      modalWindow.style.display = 'block';
+    }
+  }
+
+  close() {
+    modalWindow.style.display = 'none';
+
+    const contentToRemove = document.getElementById('modalWindow-bg');
```

Comment:
Moze wlasnie warto zamiast edytowac content, tworzy go od razu i potem tylko howac / pokazywac cale okno. Ja bylbym za takim rozwiazaniem :) Takie edycje stanu moga czasem sprawiac problemy.

Reply:
a co jak w przypadku gdy content się zmieni? Dlatego dodałem to usuwanie.

Reply:
@nowakprojects co myślisz? Okno będzie otwierane wiele razy i za każdym razem będzie inna zawartość, dlatego tak zrobiłem. Wtedy content może przyjąć obojętnie jaką formę, bo będzie całe dodane do overlay'a, a nie, że coś w nim podmieniane :-)

Reply:
@Szambelan chyba może tak być :) 

@PiotrWR Właśnie... napisaliśmy własną metodę (destory), a tutaj Paweł używa:
`Metoda ChildNode.remove() usuwa obiekt z drzewa, do którego należy.`

https://developer.mozilla.org/pl/docs/Web/API/ChildNode/remove

Reply:
Świetnie!
Ciekawe która lepsza. Sprawdzę to!

Reply:
Czyli wychodzi na to, że lepiej można było użyć samego "remove", bo działa tak samo w tym przypadku, a nie wymaga rodzica?